shared/request_caching.go

```diff
@@ -154,15 +156,34 @@ func (h cachingHandler) delegateAndCache(w http.ResponseWriter, r *http.Request,
 
 // NewCachingHandler produces a caching handler with an underlying delegate
 // handler, cache, cacheability decision function, and cache key producer.
-func NewCachingHandler(delegate http.Handler, cache ReadWritable, isCacheable func(*http.Request) bool, getCacheKey func(*http.Request) interface{}, shouldCache func(int) bool) http.Handler {
+func NewCachingHandler(ctx context.Context, delegate http.Handler, cache ReadWritable, isCacheable func(*http.Request) bool, getCacheKey func(*http.Request) interface{}, shouldCache func(int, []byte) bool) http.Handler {
```

Comment:
So, I don't think we should ever pass in nil (so that it's instead quite obvious what the value is at the call-site).